src/transforms/projectionpursuit.jl

```diff
@@ -0,0 +1,208 @@
+the standard multivariate Gaussian distribution.
+
+This iterative algorithm repeatedly finds a direction of projection `α` that maximizes a score of
+non-Gaussianity known as the projection index `I(α)`. The samples projected along `α` are then
+transformed with the [`Quantile`](@ref) transform to remove the non-Gaussian structure. The
+other coordinates in the rotated orthonormal basis `Q = [α ...]` are left untouched.
+
+The non-singularity of Q is controlled by assuring that norm(det(Q)) ≥ `tol`. The iterative 
+process terminates whenever the transformed samples are "more Gaussian" than `perc`% of `n`
+randomly generated samples from the standard multivariate Gaussian distribution, or when the 
+number of iterations reaches a maximum `maxiter`.
+
+# Examples
+
+```julia
+ProjectionPursuit()
+ProjectionPursuit(deg=10)
+ProjectionPursuit(perc=.85, n=50)
+ProjectionPursuit(tol=1e-4, maxiter=250, deg=5, perc=.95, n=100)
+```
+
+"""
+
+struct ProjectionPursuit{T} <: Stateless
+  tol::T
+  maxiter::Int
+  deg::Int
+  perc::T
+  n::Int
+end
+
+ProjectionPursuit(;tol=1e-6, maxiter=100, deg=5, perc=.95, n=100) =
+  ProjectionPursuit{typeof(tol)}(tol, maxiter, deg, perc, n)
+
+isrevertible(::Type{<:ProjectionPursuit}) = true
+
+# transforms a row of random variables into a convex combination 
+# of random variables with values in [-1,1] and standard normal distribution
+function rscore(Z, α)
+  ᾱ = (1/norm(α)) .* α
+  X = Z * ᾱ
+  2 * cdf.(Normal(), X) .- 1
+end
+
+function pindex(transform, Z, α)
+  X = rscore(Z, α)
+  I = (3/2) * mean(X)^2
+  if transform.deg > 1
+    Pⱼ₋₂, Pⱼ₋₁ = ones(size(X)), X
+    for j = 2:transform.deg
+      Pⱼ₋₂, Pⱼ₋₁ = 
+        Pⱼ₋₁, (1/j) * ((2j-1) * X .* Pⱼ₋₁ - (j-1) * Pⱼ₋₂)
+      I += ((2j+1)/2) * (mean(Pⱼ₋₁))^2
+    end
+  end
+  I
+end
+
+function optimaxis(transform, Z)
+  _, q = size(Z)
+  Iₘₐₓ, iₘₐₓ = 0, 0
+  Id = Matrix(1.0I, q, q)
+  for i = 1:q
+    Ip = pindex(transform, Z, Id[:,i])
+    if Ip > Iₘₐₓ
+      Iₘₐₓ, iₘₐₓ = Ip, i
+    end
+  end
+  Id[:,iₘₐₓ]
+end
+
+function optimdiag(transform, Z, ᾱ)
+  l  = length(ᾱ)
+  I₀ = pindex(transform, Z, ᾱ)
+  Id = Matrix(1.0I, l, l)
+  for i in 1:l
+    eᵢ = Id[:,i]
+    f₊ = pindex(transform, Z, (1/√(2+2ᾱ[i])) * (ᾱ + eᵢ))
+    f₋ = pindex(transform, Z, (1/√(2-2ᾱ[i])) * (ᾱ - eᵢ)) * (ᾱ[i] != 1)
+    f = (f₊ > f₋) * f₊ + (f₊ ≤ f₋) * f₋ 
+    s = (f₊ > f₋) * 1 + (f₊ ≤ f₋) * (-1)
+    if f > pindex(transform, Z, ᾱ)
+      ᾱ = (1/√(2+2s*ᾱ[i])) * (ᾱ + s*eᵢ)
+    end
+  end
+  pindex(transform, Z, ᾱ) ≥ I₀ ? ᾱ : optimaxis(transform, Z)
+end
+
+function optimgen(transform, Z, α₀)
+  f(a) = -pindex(transform, Z, a)
+  op = optimize(f, α₀)
+  minimizer(op)
+end
+
+function optim(transform, Z)
+  a_axis = optimaxis(transform, Z)
+  a_diag = optimdiag(transform, Z, a_axis)
+  optimgen(transform, Z, a_diag)  
+end
+
+# transforms the cdf of a convex combination of r.v. to a standard normal cdf
+function rmstructure(transform, Z, α)
+  l = length(α)
```

Comment:
Can you please confirm that the author uses the symbol l for the dimension of the problem? Papers usually choose n, m, p, k for integer variables.